docs: Explain how to make macros persist. (#2702)
KaTeX/KaTeX: docs/api.md

// '<span class="katex">...</span>'
```

## Persistent Macros

> KaTeX documentation tells the author that `\gdef` will create a macro that persists between KaTeX fields. In order to fulfill that promise, you must make certain provisions. Say that a parser has dropped from the sky with an array of DOM elements that contain math. You could then write this code:

Reviewer: Here's a proposed second draft:

```suggestion
If you want to define macros within one math element that can be used in other math elements, you need to do two things:
1. Create one shared `macros` object that you pass into every call to `katex.render` or `katex.renderToString`. (Do not create a fresh `macros` object for each call.)
2. In your math content, use `\gdef` to define your macros (so that they are defined in the global group, not just the local group of that particular math mode).

For example, suppose that you have an array `mathElements` of DOM elements that contain math. Then you could then write this code:
```

Author: > **If** you want to define macros within one math element that can be used in other math elements

I'd like to keep consistency between what we say here and what we say in  `supported.md`. `supported.md` does not have a qualifying "if". It says that `\gdef` will persist, full stop.  Since security makes that uncertain, I want to revise `supported.md` before this PR goes anywhere.

 I'll use much of your draft here, but I may revise parts of it.  Revision coming later today.

Reviewer: Sounds good.  I just found it weird to be referring to another part of the documentation, without even a link, as it's assuming a lot of the reader (what if they read this part before the other part?).  Perhaps the documentation that talks about `\gdef` should also refer to this section, so the proviso is linked directly from the potential offender.

Author: > without even a link

Excellent point. I will definitely write a link into this PR.

Not a reverse link, though. An author in, say [Observable](https://observablehq.com/), does not need to be bothered with the machinery behind the scenes. That's the sort of person I'm targeting in `supported.md`

```js
const macros = {};
for (let element of mathElements) {
    katex.render(element.textContent, element, {
        throwOnError: false,
        macros
    };
```

Notice that you create the `macros` object outside the loop. That object will contain macros that the user defines using `\gdef` and the macros will persist between calls to `katex.render`.

> 

Reviewer: It's probably worth warning about security:

```suggestion

### Security of Persistent Macros
Persistent macros can change the behavior of KaTeX (e.g. redefining standard commands), so for security, such a setup should be used only for multiple elements of common trust.  For example, you might enable persistent macros within a message posted by a single user (by creating a `macros` object for that message), but you probably should not enable persistent macros across multiple messages posted by multiple users.
```

Author: Good point. I'll add it.

The last argument to `katex.render` and `katex.renderToString` can contain
[a variety of rendering options](options.md).

## Persistent Macros

KaTeX’s [macro documentation](supported.html#gdef) tells the author that `\gdef` will create a macro that persists between KaTeX elements. In order to enable that persistence, you must create one shared `macros` object that you pass into every call to `katex.render` or `katex.renderToString`. (Do not create a fresh `macros` object for each call.)

> For example, suppose that you have an array `mathElements` of DOM elements that contain math. Then you could then write this code:

Reviewer: ```suggestion
For example, suppose that you have an array `mathElements` of DOM elements that contain math. Then you could write this code:
```
(double "then")
